mod-clans/src/main/java/gg/solarmc/loader/clans/Clan.java

```diff
@@ -0,0 +1,442 @@
+/*
+ *
+ *  * dataloader
+ *  * Copyright © 2021 SolarMC Developers
+ *  *
+ *  * dataloader is free software: you can redistribute it and/or modify
+ *  * it under the terms of the GNU Affero General Public License as
+ *  * published by the Free Software Foundation, either version 3 of the
+ *  * License, or (at your option) any later version.
+ *  *
+ *  * dataloader is distributed in the hope that it will be useful,
+ *  * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  * GNU Affero General Public License for more details.
+ *  *
+ *  * You should have received a copy of the GNU Affero General Public License
+ *  * along with dataloader. If not, see <https://www.gnu.org/licenses/>
+ *  * and navigate to version 3 of the GNU Affero General Public License.
+ *
+ */
+
+package gg.solarmc.loader.clans;/*
+ *
+ *  * dataloader
+ *  * Copyright © 2021 SolarMC Developers
+ *  *
+ *  * dataloader is free software: you can redistribute it and/or modify
+ *  * it under the terms of the GNU Affero General Public License as
+ *  * published by the Free Software Foundation, either version 3 of the
+ *  * License, or (at your option) any later version.
+ *  *
+ *  * dataloader is distributed in the hope that it will be useful,
+ *  * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  * GNU Affero General Public License for more details.
+ *  *
+ *  * You should have received a copy of the GNU Affero General Public License
+ *  * along with dataloader. If not, see <https://www.gnu.org/licenses/>
+ *  * and navigate to version 3 of the GNU Affero General Public License.
+ *
+ */
+
+import gg.solarmc.loader.SolarPlayer;
+import gg.solarmc.loader.Transaction;
+import gg.solarmc.loader.schema.tables.records.ClansClanAlliancesRecord;
+import gg.solarmc.loader.schema.tables.records.ClansClanEnemiesRecord;
+import gg.solarmc.loader.schema.tables.records.ClansClanInfoRecord;
+import gg.solarmc.loader.schema.tables.records.ClansClanMembershipRecord;
+import org.jooq.DSLContext;
+
+import java.util.Optional;
+import java.util.Set;
+
+import static gg.solarmc.loader.schema.tables.ClansClanAlliances.CLANS_CLAN_ALLIANCES;
+import static gg.solarmc.loader.schema.tables.ClansClanEnemies.CLANS_CLAN_ENEMIES;
+import static gg.solarmc.loader.schema.tables.ClansClanMembership.CLANS_CLAN_MEMBERSHIP;
+import static gg.solarmc.loader.schema.tables.ClansClanInfo.CLANS_CLAN_INFO;
+
+/**
+ * Note to reader of this source code: Come play SolarMC!
+ */
+@SuppressWarnings("unused")
+public class Clan {
+
+    private final int clanID;
+    private final String clanName;
+
+    private Set<ClanMember> members;
```

Comment:
```suggestion
    private volatile Set<ClanMember> members;
```
The pattern you are using, whereby you atomically re-assign the value of members, works best if you mark the variable as volatile.